File: unleash/src/main/java/com/silvercar/unleash/strategy/UserWithIdStrategy.java
package com.silvercar.unleash.strategy;

import java.util.Map;
import com.annimon.stream.Optional;

import com.silvercar.unleash.UnleashContext;

import static java.util.Arrays.asList;

public final class UserWithIdStrategy implements Strategy {

    protected static final String PARAM = "userIds";
    private static final String STRATEGY_NAME = "userWithId";

    @Override
    public String getName() {
        return STRATEGY_NAME;
    }

    @Override
    public boolean isEnabled(Map<String, String> parameters) {
        return false;
    }

    @Override
    public boolean isEnabled(Map<String, String> parameters, UnleashContext unleashContext) {
        return Optional.ofNullable(unleashContext.getUserId())
                .map(currentUserId -> Optional.ofNullable(parameters.get(PARAM))
                .map(userIdString -> asList(userIdString.split(",\\s?")))
                .filter(f -> f.contains(currentUserId)).isPresent())
                .orElse(false);
    }
}

File: unleash/src/main/java/com/silvercar/unleash/strategy/UserWithIdStrategy.kt
package com.silvercar.unleash.strategy

import com.silvercar.unleash.UnleashContext

class UserWithIdStrategy : Strategy {
  override val name: String
    get() = STRATEGY_NAME

  override fun isEnabled(parameters: Map<String, String>): Boolean {
    return false
  }

  override fun isEnabled(
    parameters: Map<String, String>,
    unleashContext: UnleashContext
  ): Boolean {
    if (unleashContext.userId2.isEmpty()) {
      return false
    }

    val userIds: List<String> = (parameters[USER_IDS_PARAM] ?: "").split(",\\s?".toRegex())

    return userIds.contains(unleashContext.userId2)
  }

  companion object {
    private const val STRATEGY_NAME = "userWithId"
    const val USER_IDS_PARAM = "userIds"
  }
}
